create-app.ts

```diff
@@ -0,0 +1,320 @@
+import { shouldUseYarn } from './helpers/should-use-yarn'
+import { isWriteable } from './helpers/is-writeable'
+
+export class DownloadError extends Error {}
+
+export async function createApp({
+  appPath,
+  useNpm,
+  example,
+  examplePath,
+  typescript,
+}: {
+  appPath: string
+  useNpm?: boolean
+  example?: string
+  examplePath?: string
+  typescript?: boolean
+}): Promise<void> {
+  let repoInfo: RepoInfo | undefined
+  const template = typescript ? 'typescript' : 'default'
+  example = 'https://github.com/henripal/create-dao/tree/main/template'
+
+  if (example) {
+    let repoUrl: URL | undefined
+
+    try {
+      repoUrl = new URL(example)
+    } catch (error: any) {
+      if (error.code !== 'ERR_INVALID_URL') {
+        console.error(error)
+        process.exit(1)
+      }
+    }
+
+    if (repoUrl) {
+      if (repoUrl.origin !== 'https://github.com') {
+        console.error(
+          `Invalid URL: ${chalk.red(
+            `"${example}"`
+          )}. Only GitHub repositories are supported. Please use a GitHub URL and try again.`
+        )
+        process.exit(1)
+      }
+
+      repoInfo = await getRepoInfo(repoUrl, examplePath)
+
+      if (!repoInfo) {
+        console.error(
+          `Found invalid GitHub URL: ${chalk.red(
+            `"${example}"`
+          )}. Please fix the URL and try again.`
+        )
+        process.exit(1)
+      }
+
+      const found = await hasRepo(repoInfo)
+
+      if (!found) {
+        console.error(
+          `Could not locate the repository for ${chalk.red(
+            `"${example}"`
+          )}. Please check that the repository exists and try again.`
+        )
+        process.exit(1)
+      }
+    } else if (example !== '__internal-testing-retry') {
+      const found = await hasExample(example)
+
+      if (!found) {
+        console.error(
+          `Could not locate an example named ${chalk.red(
+            `"${example}"`
+          )}. It could be due to the following:\n`,
+          `1. Your spelling of example ${chalk.red(
+            `"${example}"`
+          )} might be incorrect.\n`,
+          `2. You might not be connected to the internet.`
+        )
+        process.exit(1)
+      }
+    }
+  }
+
+  const root = path.resolve(appPath)
+
+  if (!(await isWriteable(path.dirname(root)))) {
+    console.error(
+      'The application path is not writable, please check folder permissions and try again.'
+    )
+    console.error(
+      'It is likely you do not have write permissions for this folder.'
+    )
+    process.exit(1)
+  }
+
+  const appName = path.basename(root)
+
+  await makeDir(root)
+  if (!isFolderEmpty(root, appName)) {
+    process.exit(1)
+  }
+
+  const useYarn = useNpm ? false : shouldUseYarn()
+  const isOnline = !useYarn || (await getOnline())
+  const originalDirectory = process.cwd()
+
+  const displayedCommand = useYarn ? 'yarn' : 'npm'
+  console.log(`Creating a new Next.js app in ${chalk.green(root)}.`)
+  console.log()
```

Comment:
and remove this 😁

Reply:
Actually, I see this used in several places. Is it for a `\n` style effect?

Reply:
> Is it for a \n style effect?

I think so, yes

Reply:
yes! don't have a strong opinon there